### PR TITLE
Update GHC versions in CI

### PR DESCRIPTION
**Pull request by nitinprakash96**

### CODE REVIEW

**Comment by turboMaCk** [comment]
seems there is some issue with yaml:

```
Stack could not load and parse
       /home/runner/work/postgresql-simple-named/postgresql-simple-named/--test as a YAML
       configuraton file.
```

File: .github/workflows/haskell-ci.yml [hunk 1]
```diff
@@ -56,10 +56,9 @@ jobs:
     strategy:
       matrix:
         stack:
-          - 2.7.5
-        stack-yaml:
-          - stack-8.10.7.yaml
-          - stack-9.0.2.yaml
+          - 2.11.1
+        ghc:
```
Comment (michivi) [inline .github/workflows/haskell-ci.yml]: Is there a specific reason why you removed the old stack.yaml files? We've lost the guarantee that the library would work with older Stackage snapshots.

Comment (jhrcek) [reply .github/workflows/haskell-ci.yml]: Yea, I'd say it's pretty cheap to keep, say 2-3 ghc minor versions in CI and there's almost no maintenance overhead. How about keeping at least 2 stack yamls, one with ghc 9.4.5 and one with 9.2.8?

Comment (nitinprakash96) [reply .github/workflows/haskell-ci.yml]: @michivi I just wanted to have a single `stack` file. I meant to bring both cabal and stack to the same page by passing `ghc` field. As for guarantees with older version, I'd just add more versions to `ghc` field like @jhrcek suggests.

Comment (jhrcek) [reply .github/workflows/haskell-ci.yml]: I think it you want to combine using stack.yaml (which determines which stack-installed ghc will be used) and externally (by ghcup) installed ghc, you would have to use `stack build ... --system-ghc`.

But I find that unnecessarily complicated.

I'd prefer just having 2-3 stack yamls and passing those as matrix parameters + not specifying ghc versions as matrix parameters. That feels like more "stack native" way to do it.

Comment (michivi) [reply .github/workflows/haskell-ci.yml]: Yeah, if you're using a stackage LTS version, then you have an associated GHC version requirement. Using a different version than what is specified in your current resolver will most likely bring issue (unsatisfied constraints on `base`). Using the `matrix` parameters and different stack.yaml seems preferable.

Comment (jhrcek) [reply .github/workflows/haskell-ci.yml]: As an illustration that current approach doesn't work: your stack 9.0.2 job was actually using ghc 9.4.5

Screenshot from 2023-06-29 07-48-54

File: stack-9.0.2.yaml [hunk 2]
```diff
@@ -0,0 +1 @@
+resolver: lts-19.31
```
Comment (jhrcek) [inline stack-9.0.2.yaml]: Let's use latest stackage snapshot per each ghc version (see "Latest releases per GHC version" on [stackage.org](https://www.stackage.org/))

For 9.0.2 it's lts-19.33, in other config you're using latest already.

**Review by jhrcek — APPROVED** [review]
LGTM

**Pull request merged**